reframe/core/schedulers/slurm.py

```diff
@@ -88,6 +89,11 @@ def __init__(self):
         self._is_cancelling = False
         self._is_job_array = None
         self._update_state_count = 0
+        self._completion_time = None
+
+    @property
+    def completion_time(self):
```

Comment:
Allow this to get the `job` as argument (so no more property), and leave `_update_state` untouched.